lms/djangoapps/email_marketing/signals.py

```diff
@@ -54,41 +55,33 @@ def add_email_marketing_cookies(sender, response=None, user=None,
         if sailthru_content:
             post_parms['cookies'] = {'anonymous_interest': sailthru_content}
 
-    try:
-        sailthru_client = SailthruClient(email_config.sailthru_key, email_config.sailthru_secret)
-        log.info(
-            'Sending to Sailthru the user interest cookie [%s] for user [%s]',
-            post_parms.get('cookies', ''),
-            user.email
-        )
-        time_before_call = datetime.datetime.now()
+    time_before_call = datetime.datetime.now()
+    sailthru_response = get_email_cookies_via_sailthru.delay(user.email, post_parms)
 
-        sailthru_response = \
-            sailthru_client.api_post("user", post_parms)
+    try:
+        # synchronous call to celery task with a timeout limit
+        sailthru_response.get(timeout=email_config.user_registration_cookie_timeout_delay,
+                              propagate=True)
+        cookie = sailthru_response.result
 
+    except TimeoutError as exc:
+        log.error("Timeout error while attempting to obtain cookie from Sailthru: %s", unicode(exc))
```

Comment:
> Iam using docker for development, so i dont know for now how to pass --setting=devstack_with_worker for server startup. Secondly tested this code in a sandbox and it was working fine.

Ah, okay I understand. If things are working on a sandbox you'll be fine (if you're still worried about deadlock situations, use incognito windows for manual testing or something similar to https://github.com/edx/edx-load-tests to do it programatically - my recommendation is to do it manually). Docker devstack is single-process just like vagrant devstack, so the failures there are not unexpected.